### DISCUSSION TITLE
how to get image url?

### DISCUSSION THREAD

**Discussion by sindhubha** [fastapi/fastapi#9430] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [x] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/get_company_list/")
async def get_company_list(request: Request,
                           company_id: int = None,
                           cnx: Session = Depends(get_db)):
    try:
        base_path = os.path.abspath(os.path.dirname(__file__))
        where = ""
        if company_id is not None:
            where = f" and mc.company_id = {company_id}"
        query = text(f'''
                    SELECT
                        mc.*,
                        CONCAT('http://{base_path}/attachment/company_logo/',mc.logo) AS logo,
                         FROM
                        master_company as mc
                           WHERE mc.status='active' {where}
                    ''')
        data = cnx.execute(query).mappings().all()

        # return data without decoding image
        results = []
        for i in data:
            new_logo = dict(i)
            if new_logo["logo"] is not None:
                new_logo["logo"] = "http://" + os.path.relpath(new_logo['logo'], start=base_path).replace("\\", "/")
            results.append(new_logo)
```

### Description

 save the image as an file that image file is stored in the attachment folder i need to get the image list (image show  as an url )
im not using cloud . is there any option to get the image as an url 

### Operating System

Windows

### FastAPI Version

0.95.0

### Python Version

3.9.12

**Comment by notypecheck** [comment]
You usually store image path (e.g. `/path/to/image/some-image.png`) and then concatenate it with your server address (for example `https://my-site.com/media`), ideally you'd use something like nginx to serve your image files.

**Reply by sindhubha** [reply]
without using  server address because different users using different server. not using any cloud service

**Reply by notypecheck** [reply]
By user you mean a user of your application or an instance of it?

**Reply by sindhubha** [reply]
in frontend they want to get image from backend side. incase frontend and backend using different server,  then  how to get the image  from the backend side

**Comment by Viicos** [comment]
Unrelated but it seems that your code is vulnerable to [SQL injection](https://docs.python.org/3/library/sqlite3.html#sqlite3-placeholders)

**Reply by notypecheck** [reply]
Yep 👀

**Comment by deniselupe** [comment]
Hello, I'm not sure if you're actually needing it as a URL or if you just need to fetch the image so that you can display it on the frontend. 

If the hostname of the server changes, but you know where the image is located, you can just return the file itself from the backend using something like `FileResponse`, and then on the client side you can display the image directly to your web page. 

Here is an example from a small application I wrote. The user uploads an image, and then I send it right back to be displayed on the web page.

```
# backend
@app.post("/img")
async def upload_img(uploaded_file: UploadFile):
    file_location = f"./uploaded_files/{uploaded_file.filename}"
    file_content = await uploaded_file.read()

    with open(file_location, "wb+") as fname:
        fname.write(file_content)

    return FileResponse(file_location)
```

```
# frontend - using React
  const sendAndRenderFile = (formData) => {
    axios.post("http://127.0.0.1:8000/img", formData, {responseType: 'blob'})
      .then(res => {
        setImage(URL.createObjectURL(res.data));
      })
  };
```

Hopefully, this helps!

**Comment by abhint** [comment]
```py
import os
from fastapi import FastAPI, UploadFile, File

app = FastAPI()
UPLOAD_FOLDER = "attachments"

@app.post("/upload_image/")
async def upload_image(image: UploadFile = File(...)):
    # Save the uploaded image to the specified folder
    file_path = os.path.join(UPLOAD_FOLDER, image.filename)
    with open(file_path, "wb") as file:
        file.write(await image.read())

    # Construct the URL of the uploaded image
    base_url = "https://yourdomain.com"
    image_url = f"{base_url}/{file_path}"

    return {"image_url": image_url}

@app.get("/image_list/")
async def get_image_list():
    # Get a list of image files in the attachments folder
    image_list = []
    for filename in os.listdir(UPLOAD_FOLDER):
        if filename.endswith((".jpg", ".jpeg", ".png", ".gif")):
            file_path = os.path.join(UPLOAD_FOLDER, filename)
            image_url = f"https://yourdomain.com/{file_path}"
            image_list.append(image_url)

    return {"image_list": image_list}

```

**Reply by MokhtarOmar16** [reply]
i think we can use `pydantic` more and also use `Request`  object to got domain name
